metadata/src/main/java/org/apache/kafka/metadata/KafkaConfigSchema.java

```diff
@@ -244,4 +245,34 @@ private ConfigEntry toConfigEntry(ConfigDef.ConfigKey configKey,
             translateConfigType(configKey.type()),
             configKey.documentation);
     }
+
+    /**
+     * OrderedConfigResolver helps to find the configs in the order of the list config maps.
+     * One thing to notice that, when calling containsKey, if a config contains a null value entry,
+     * it will return false as null value means the config value should be ignored.
+     **/
+    public static class OrderedConfigResolver {
+        List<Map<String, ?>> configs;
+        public OrderedConfigResolver(List<Map<String, ?>> maps) {
+            configs = maps;
+        }
+
+        public OrderedConfigResolver(Map<String, ?> map) {
+            configs = new ArrayList<>();
+            configs.add(map);
+        }
+        public boolean containsKey(String key) {
+            for (Map<String, ?> config : configs) {
+                if (config.containsKey(key)) return config.get(key) != null;
```

Comment:
If a later config map has a non-null, I think we could mistakenly return false here.